src/senselab/audio/tasks/speech_enhancement/speechbrain.py

```diff
@@ -0,0 +1,92 @@
+"""This module provides the Speechbrain interface for speech enhancement."""
+from typing import Dict, List, Optional
+
+import torch
+from speechbrain.inference.separation import SepformerSeparation as separator
+
+from senselab.audio.data_structures.audio import Audio
+from senselab.utils.data_structures.device import DeviceType, _select_device_and_dtype
+from senselab.utils.data_structures.model import HFModel
+
+
+class SpeechBrainEnhancer:
+    """A factory for managing SpeechBrain enhancement pipelines."""
+
+    _models: Dict[str, separator] = {}
+
+    @classmethod
+    def _get_speechbrain_model(
+        cls,
+        model: HFModel,
+        device: Optional[DeviceType] = None,
+    ) -> separator:
+        """Get or create a SpeechBrain enhancement model.
+
+        Args:
+            model (HFModel): The SpeechBrain model.
+            device (Optional[DeviceType]): The device to run the model on.
+                Only CPU and CUDA are supported.
+
+        Returns:
+            separator: The SpeechBrain enhancement model.
+        """
+        device, _ = _select_device_and_dtype(
+            user_preference=device, compatible_devices=[DeviceType.CUDA, DeviceType.CPU]
+        )
+        key = f"{model.path_or_uri}-{model.revision}-{device.value}"
+        if key not in cls._models:
+            cls._models[key] = separator.from_hparams(
+                source=model.path_or_uri, 
+                run_opts={"device": device.value}
+            )
+        return cls._models[key]
+
+    @classmethod
+    def enhance_audios_with_speechbrain(
+        cls,
+        audios: List[Audio],
+        model: HFModel = HFModel(path_or_uri="speechbrain/sepformer-wham16k-enhancement", revision="main"),
```

Comment:
I guess similar to above. We should just be consistent on what we use.

Reply:
since speechbrain models are HF models, there is no need to initialize model as SenselabModel. We want to initialize it as HFModel to better validate the input indeed